packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -0,0 +1,278 @@
+    if (currentSelection.length === 0) {
+      return <>{placeholder}</>;
+    }
+    const leftoverCount = currentSelection.length - 1;
+    if (leftoverCount === 0) {
+      return (
+        <span className={styles.currentSelection}>{currentSelection[0]}</span>
+      );
+    }
+    return (
+      <div>
+        <span className={styles.currentSelection}>{currentSelection[0]}</span>{' '}
+        and {leftoverCount} more
+      </div>
+    );
+  }, [currentSelection, placeholder, styles.currentSelection]);
+
+  const childrenLength = React.Children.count(children);
+
+  return (
+    <div
+      data-test-id={testId}
+      className={cx(styles.multiselect, className)}
+      ref={ref}
+    >
+      <Popover
+        usePortal={usePortal}
+        isOpen={isOpen}
+        onClose={() => setIsOpen(false)}
+        isFullWidth={listWidth === 'full'}
+        renderOnlyWhenOpen={false}
+      >
+        <Popover.Trigger>
+          <Button
+            aria-label="open Select"
+            ref={toggleRef}
+            onClick={() => setIsOpen(!isOpen)}
+            startIcon={startIcon}
+            endIcon={<ChevronDownIcon />}
+          >
+            {renderMultiselectLabel()}
+          </Button>
+        </Popover.Trigger>
+        <Popover.Content
+          ref={listRef}
+          className={styles.content(listMaxHeight)}
+          testId="cf-multiselect-container"
+        >
+          <>
+            {hasSearch && (
+              <>
+                <TextInput
+                  type="text"
+                  value={searchValue}
+                  className={styles.inputField}
+                  id={id}
+                  data-test-id="cf-multiselect-search"
+                  placeholder={searchPlaceholder}
+                  onChange={(event) => {
+                    handleSearchChange(event);
+                  }}
+                  ref={mergeRefs(searchInputRef, inputRef)}
+                />
+                <IconButton
+                  aria-label="Clear"
+                  className={styles.toggleButton}
+                  variant="transparent"
+                  icon={
+                    searchValue ? (
+                      <CloseIcon variant="muted" />
+                    ) : (
+                      <SearchIcon variant="muted" />
+                    )
+                  }
+                  onClick={() => {
+                    if (searchValue) {
+                      resetSearch();
+                    }
+                  }}
+                  isDisabled={searchValue === ''}
+                  size="small"
+                />
+              </>
+            )}
+            {isLoading && <ListItemLoadingState />}
+
+            {!isLoading && childrenLength > 0 && (
+              <ul className={styles.list} data-test-id="cf-multiselect-items">
+                {React.Children.map(children, (child) => {
+                  if (React.isValidElement(child)) {
+                    return React.cloneElement(child, {
```

Comment:
For the search input value if there is one, because it is used to highlight the search phrase in the found elements.

For optimisation i could only clone when there is a search and if not, just return the children without the enrichment. 
I have to change it also a bit, once i allow also grouping, because then there would not only be Options as child but also something for the group title.

There probaply should also be a more specific type for children then just a ReactNode then i guess. To make clear that it expects either an Option or a GroupTitle.